### PR TITLE
DEVEXP-551: Support additional properties onto Start request

### PR DESCRIPTION
**Pull request by JPPortier**

### CODE REVIEW

File: openapi-contracts/src/test/java/com/sinch/sdk/domains/verification/models/dto/v1/start/request/internal/VerificationStartSmsOptionsTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    JSONAssert.assertEquals(jsonVerificationStartSmsOptions, serializedString, true);
+  }
+
+  @Test
+  void deserializeStartVerificationSmsOptions() throws JsonProcessingException {
+    Object deserialized =
+        objectMapper.readValue(jsonVerificationStartSmsOptions, VerificationStartSmsOptions.class);
+
+    TestHelpers.recursiveEquals(deserialized, startVerificationSmsOptionDto);
```
Comment (asein-sinch) [inline openapi-contracts/src/test/java/com/sinch/sdk/domains/verification/models/dto/v1/start/request/internal/VerificationStartSmsOptionsTest.java]: Nit: I know that we are testing the equality of 2 objects and the equality is bijective, but here the order is different from what is done with JSONAssert.assertEquals(expected, actual)

File: openapi-contracts/src/test/java/com/sinch/sdk/domains/verification/models/dto/v1/start/request/internal/VerificationStartSmsOptionsTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    // acceptLanguage do not have to be serialized
+    JSONAssert.assertEquals(jsonVerificationStartSmsOptions, serializedString, true);
+  }
+
+  @Test
+  void deserializeStartSmsCallWithAcceptLanguage() throws JsonProcessingException {
+    Object deserialized =
+        objectMapper.readValue(jsonVerificationStartSmsOptions, VerificationStartSmsOptions.class);
+
+    // we are not expecting acceptLanguage to be unserialized because of not sent to network
```
Comment (asein-sinch) [inline openapi-contracts/src/test/java/com/sinch/sdk/domains/verification/models/dto/v1/start/request/internal/VerificationStartSmsOptionsTest.java]: Well, it's sent over the network, but not in the body 😉

File: openapi-contracts/src/test/resources/domains/verification/v1/start/request/VerificationStartRequestSmsDto.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
   "smsOptions": {
     "expiry": "01:02:03",
     "codeType": "Alphanumeric",
-    "template": "My template require to use '{{CODE}}' code"
+    "template": "My template require to use '{{CODE}}' code",
```
Comment (asein-sinch) [inline openapi-contracts/src/test/resources/domains/verification/v1/start/request/VerificationStartRequestSmsDto.json:12]: Nit: This is very unlikely a real-life message but at least let's correct the error
```suggestion
    "template": "My template requires to use {{CODE}} placeholder",
```

File: sample-app/src/main/java/com/sinch/sample/webhooks/verification/VerificationService.java [hunk 4]
```diff
@@ -27,7 +27,7 @@ public VerificationRequestEventResponse verificationEvent(VerificationRequestEve
     var method = event.getMethod();
 
     if (MethodEnum.SMS.equals(method)) {
-      builder = VerificationRequestEventResponseSms.builder().setCode("1234");
+      builder = VerificationRequestEventResponseSms.builder();
```
Comment (asein-sinch) [inline sample-app/src/main/java/com/sinch/sample/webhooks/verification/VerificationService.java:30]: 🤐

**Review by asein-sinch — APPROVED** [review]

File: openapi-contracts/src/test/java/com/sinch/sdk/domains/verification/models/dto/v1/start/request/internal/VerificationStartSmsOptionsTest.java (hunk 1 above)
Comment (JPPortier) [reply openapi-contracts/src/test/java/com/sinch/sdk/domains/verification/models/dto/v1/start/request/internal/VerificationStartSmsOptionsTest.java]: Right: a pattern is a pattern :-)
Fixed

File: openapi-contracts/src/test/java/com/sinch/sdk/domains/verification/models/dto/v1/start/request/internal/VerificationStartSmsOptionsTest.java (hunk 2 above)
Comment (JPPortier) [reply openapi-contracts/src/test/java/com/sinch/sdk/domains/verification/models/dto/v1/start/request/internal/VerificationStartSmsOptionsTest.java]: Yes: if comments are present to  to explain something: it have sense to explain right
Done

File: openapi-contracts/src/test/resources/domains/verification/v1/start/request/VerificationStartRequestSmsDto.json (hunk 3 above)
Comment (JPPortier) [reply openapi-contracts/src/test/resources/domains/verification/v1/start/request/VerificationStartRequestSmsDto.json:12]: Done

**Pull request merged**